### PR TITLE
Network.NIC: Remove --ids from the parameters

### PR DESCRIPTION
**Pull request by MyronFanQiu**
---
Fixes https://github.com/Azure/azure-cli/issues/9861

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by MyronFanQiu** [comment]
@tjprescott I'm not sure whether this is a right way to fix this issue. In my understanding, should we do some fix in the azure.cli.core so that the `--ids` will not show up?

**Comment by MyronFanQiu** [comment]
Besides targeting at this issue, please help me check, for a common scenario that we need to do some fix for an existed command, what do I miss? In this PR, I change the logic, add test and new recording, update the history.rst. Is there something I also need to do?

File: src/azure-cli/HISTORY.rst [hunk 1]
```diff
@@ -3,6 +3,10 @@
 Release History
 ===============
 
+**Network**
+
+* Fixed issue #9861. Remove `--ids` from the parameters.
```
Comment (tjprescott) [inline src/azure-cli/HISTORY.rst]: The network section should be located alphabetically. Also, the typical format would be:

* `network nic ip-config add`: Fixed issue where `--ids` was inadvertently exposed but did not work. 

Also, were it not for the fact that `--ids` caused a crash, we could not simply remove this. We would need to deprecate it using: https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#deprecating-commands-and-arguments

**Comment by tjprescott** [comment]
> By prefacing your commits with “Fixes”, “Fixed”, “Fix”, “Closes”, “Closed”, or “Close” when the commit is merged into master, it will also automatically close the issue.
> 
> References make it possible to deeply connect the work being done with the bug being tracked, and are a great way to add visibility into the history of your project.

From: https://guides.github.com/features/issues/#notifications

In short, use "Fixes #9861" instead of "Fixed issue #9861" to get the auto-close functionality.

File: src/azure-cli/HISTORY.rst (hunk 1 above)
Comment (MyronFanQiu) [reply src/azure-cli/HISTORY.rst]: Actually, I don't understand what is the `--ids`. Is it by design to show the `--ids` when you add `id_parts=xxx` into the `c.arguments`? What do we need as the value for this field?
I will fix other comments, thanks very much!~

Comment (tjprescott) [reply src/azure-cli/HISTORY.rst]: It is discussed here:
https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#supporting-the-ids-parameter

Comment (MyronFanQiu) [reply src/azure-cli/HISTORY.rst]: Yes. I checked that doc before.  Maybe I misunderstand something, please correct me. To be simple, what should we used as an input for the `--ids` in the example below? `/subscription/xxxx-xxxx-xxxx/resoureceGroup/myResourceGroup/Microsoft.NIC/myNIC` or `/subscription/xxxx-xxxx-xxxx/resoureceGroup/myResourceGroup/Microsoft.NIC/myNIC/publicIP/mypublicIP` or both of them?
```
Resource Id Arguments
    --ids              : One or more resource IDs. If provided, no other 'Resource Id' arguments
                         should be specified.
    --name -n          : The IP config name.
    --nic-name         : The NIC name.
    --resource-group -g: Name of resource group.
```

Comment (tjprescott) [reply src/azure-cli/HISTORY.rst]: Neither, but that is why `--ids` was being removed from the command. `--ids` should always refer to whatever the argument group's central item is.  In this case it is questionable or confusing what `--ids` should refer to, which is why it shouldn't be exposed.

**Pull request merged**